src/main/java/net/minecraftforge/fml/client/FMLClientHandler.java

```diff
@@ -342,17 +347,19 @@ public void finishMinecraftLoading()
         {
             Loader.instance().initializeMods();
         }
-        catch (CustomModLoadingErrorDisplayException custom)
-        {
-            FMLLog.log.error("A custom exception was thrown by a mod, the game will now halt", custom);
-            customError = custom;
-            SplashProgress.finish();
-            return;
-        }
         catch (LoaderException le)
         {
-            haltGame("There was a severe problem during mod loading that has caused the game to fail", le);
-            return;
+            if (le.getCause() instanceof CustomModLoadingErrorDisplayException)
```

Comment:
If you're going to mess with it, you may as well leave it better than it was before.